deploy-commands.js

```diff
@@ -0,0 +1,36 @@
+require('dotenv').config();
+
+const token = process.env.DISCORD_TOKEN;
+const clientId = process.env.CLIENT_ID;
+const guildId = process.env.GUILD_ID;
+
+const { REST } = require('@discordjs/rest');
+const { Routes } = require('discord-api-types/v9');
+const fs = require('fs');
+
+const commands = [];
+const commandFiles = fs
+	.readdirSync('./commands')
+	.filter(file => file.endsWith('.js'));
+
+for (const file of commandFiles) {
+	const command = require(`./commands/${file}`);
+	commands.push(command.data.toJSON());
+}
+
+const rest = new REST({ version: '9' }).setToken(token);
+
+(async () => {
+	try {
+		console.log('Started refreshing application (/) commands.');
+
+		await rest.put(
+			Routes.applicationGuildCommands(clientId, guildId),
```

Comment:
To deploy slash commands we need to run `deploy-commands.js`. Currently we are deploying to the test server (line 26) but on deployment we will need to run this with `Routes.applicationCommands` to push the slash commands globally (for all guilds).

I'm not sure what the best idea is to make sure that we do this for each new command; i.e. in development we should push to to the test server, but in prod we should be deploying globally (but only need to do so if command files have changed).

Any thoughts?

Reply:
As discussed we can run this on startup, checking `NODE_ENV`.

Reply:
I've moved this to the `yarn start` command, instead.